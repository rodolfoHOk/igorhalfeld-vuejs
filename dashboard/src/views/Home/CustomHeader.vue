<template>
  <h1>Custom header</h1>
</template>

<script>
export default {};
</script>
