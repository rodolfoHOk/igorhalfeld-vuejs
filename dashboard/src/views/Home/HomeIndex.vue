<template>
  <div>
    <home-header @create-account="handleAccountCreate" @login="handleLogin" />
    <home-contact />
    <div class="flex justify-center py-10 bg-brand-gray">
      <p class="font-medium text-center text-gray-800">feedbacker Â© 2023</p>
    </div>
  </div>
</template>

<script>
import { onMounted } from 'vue';
import { useRouter } from 'vue-router';
import HomeHeader from './HomeHeader.vue';
import HomeContact from './HomeContact.vue';
import useModal from '../../hooks/useModal';

export default {
  components: { HomeHeader, HomeContact },
  setup() {
    const router = useRouter();
    const modal = useModal();

    onMounted(() => {
      const token = localStorage.getItem('token');
      if (token) {
        router.push({ name: 'Feedbacks' });
      }
    });

    function handleLogin() {
      modal.open({
        component: 'ModalLogin',
      });
    }
    function handleAccountCreate() {
      modal.open({
        component: 'ModalCreateAccount',
      });
    }

    return {
      handleLogin,
      handleAccountCreate,
    };
  },
};
</script>
