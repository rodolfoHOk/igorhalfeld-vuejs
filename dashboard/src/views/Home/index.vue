<!-- eslint-disable vue/multi-word-component-names -->
<template>
  <div>
    <h1 class="text-brand-main font-bold text-lg">Home</h1>
  </div>
</template>

<script>
export default {};
</script>
