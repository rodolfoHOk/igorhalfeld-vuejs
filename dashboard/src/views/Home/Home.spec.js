import { createRouter, createWebHistory } from 'vue-router';
import { shallowMount } from '@vue/test-utils';

import Home from './HomeIndex.vue';
import { routes } from '@/router';

const router = createRouter({
  history: createWebHistory('/'),
  routes,
});

describe('<Home />', () => {
  it('should render Home correctly', async () => {
    router.push('/');
    await router.isReady();

    const wrapper = shallowMount(Home, {
      global: {
        plugins: [router],
      },
    });

    expect(wrapper.html()).toMatchSnapshot();
  });
});
