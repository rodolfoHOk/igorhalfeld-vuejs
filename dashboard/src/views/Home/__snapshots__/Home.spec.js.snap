// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Home /> should render Home correctly 1`] = `
<div>
  <home-header-stub></home-header-stub>
  <home-contact-stub></home-contact-stub>
  <div class="flex justify-center py-10 bg-brand-gray">
    <p class="font-medium text-center text-gray-800">feedbacker Â© 2023</p>
  </div>
</div>
`;
